app/src/org/commcare/activities/LoginActivityUiController.java

```diff
@@ -194,17 +224,28 @@ public void refreshView() {
 
         // Decide whether or not to show the app selection spinner based upon # of usable apps
         ArrayList<ApplicationRecord> readyApps = MultipleAppsUtil.getUsableAppRecords();
+        boolean promptIncluded = false;
         ApplicationRecord presetAppRecord = getPresetAppRecord(readyApps);
-        if (readyApps.size() == 1 || presetAppRecord != null) {
+        if ((readyApps.size() == 1 && (!ConnectIdManager.isConnectIdIntroduced() || ConnectIdManager.isUnlocked()))
+                || presetAppRecord != null) {
+            setLoginInputsVisibility(true);
```

Comment:
@OrangeAndGreen I just noticed this while working on some merge conflicts, But since this code also gets triggered when connect is not enabled, should not ` setLoginInputsVisibility(true);` should be in the `if` later on with condition `  if (ConnectIdManager.isUnlocked()) {` ?